dispatch/static/manager/src/js/components/ContentEditor/embeds/VideoEmbed.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+
+import { FormInput, TextInput } from '../../inputs'
+
+class VideoEmbedComponent extends React.Component {
+
+  constructor(props){
+    super(props)
+
+    this.handleUrlChange = this.handleUrlChange.bind(this)
+    this.insertVideo = this.insertVideo.bind(this)
+
+    this.state = {
+      inserted: false,
+      validURL: false
+    }
+  }
+
+  handleUrlChange(e){
+    e.preventDefault()
+    this.props.updateField('url', e.target.value)
+    console.log("URL IS: "+this.props.data.url);
+
+  }
+
+
+  insertVideo(){
+    var id = this.validYouTube(this.props.data.url);
+    this.props.updateField('id',id)
+    console.log("ID IS: "+this.props.data.id)
+
+    if(this.props.data.id){
+      this.setState({
+        validURL: true,
+        inserted : true
+      })
+    }
+  }
+
+  validYouTube(url){
+     var p = /^(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?(?=.*v=((\w|-){11}))(?:\S+)?$/
+     return (url.match(p)) ? RegExp.$1 : false
+  }
+
+
+  renderInput(){
+    return(
+      <div>
+        <form>
+          <FormInput label='URL'>
+            <TextInput
+              fill={true}
+              value={this.props.data.url}
+              onChange={this.handleUrlChange} />
+          </FormInput>
+        </form>
+        <button className="c-input--button"  onClick={this.insertVideo}>Insert</button>
```

Comment:
You can use the `Button` component here:

```jsx

import { Button } from '@blueprintjs/core'

...

<Button onClick={e => this.insertVideo()}>Insert</Button>
```

I also want to move away from binding methods in the constructor (it just looks ugly IMO). If you use the function shorthand as shown above, you can remove `this.insertVideo = this.insertVideo.bind(this)` from the constructor.